kernel/mailbox.c

```diff
@@ -447,3 +455,25 @@ int k_mbox_get(struct k_mbox *mbox, struct k_mbox_msg *rx_msg, void *buffer,
 
 	return result;
 }
+
+#ifdef CONFIG_OBJ_CORE_MAILBOX
+
+static int init_mailbox_module(void)
+{
+	/* Initialize mailbox object type */
+
+	z_obj_type_init(&obj_type_mailbox, K_OBJ_TYPE_MBOX_ID,
+			offsetof(struct k_mbox, obj_core));
+
+	/* Initialize and link satically defined mailboxes */
+
+	STRUCT_SECTION_FOREACH(k_mbox, mbox) {
+		k_obj_core_init_and_link(K_OBJ_CORE(mbox), &obj_type_mailbox);
+	}
```

Comment:
I had another comment on one of these but can't find it.  I'm realizing that.

1. This **exact** init code is being repeated verbatim for every object type, the only difference being the type-specific stuff (i.e. the pointer to the type object, the type ID, the name-of the section).

2. There's no actual type-specific logic being implemented in any of them, at all.  They're all just initializing the type struct and constructing a runtime list of the static objects.

3. All that info is known a-priori at build time already.

That's a ton of boiler plate code and duplicated instruction bytes.  Can't you construct a table at build time such that a single initialization pass can be done from just one function?

Reply:
This is a good idea. I will create an enhancement so this does not get lost.